test/framework/src/main/java/org/elasticsearch/test/engine/MockEngineSupport.java

```diff
@@ -145,8 +151,19 @@ public AssertingIndexSearcher newSearcher(Engine.Searcher searcher) throws Engin
         if (reader instanceof DirectoryReader && mockContext.wrapReader) {
             wrappedReader = wrapReader((DirectoryReader) reader);
         }
+        final IndexSearcher delegate = new IndexSearcher(wrappedReader);
+        delegate.setSimilarity(searcher.searcher().getSimilarity());
+        delegate.setQueryCache(filterCache);
+        delegate.setQueryCachingPolicy(filterCachingPolicy);
+        final XIndexSearcher wrappedSearcher = new XIndexSearcher(delegate);
         // this executes basic query checks and asserts that weights are normalized only once etc.
-        final AssertingIndexSearcher assertingIndexSearcher = new AssertingIndexSearcher(mockContext.random, wrappedReader);
+        final AssertingIndexSearcher assertingIndexSearcher = new AssertingIndexSearcher(mockContext.random, wrappedReader) {
+            @Override
+            protected void search(List<LeafReaderContext> leaves, Weight weight, Collector collector) throws IOException {
+                // we cannot use the asserting searcher because the weight is created by the ContextIndexSearcher
+                wrappedSearcher.search(leaves, weight, collector);
+            }
```

Comment:
I understand we don't want to call `AssertingIndexSearcher::search` as it will fail, as it expects `AssertingWeight`.
But I am wondering if we are losing anything by not calling `search` method of `AssertingIndexSearcher`?

Reply:
Tricky tricky, I can't think of a way to work around this?

Reply:
Yep me neither, it's not a regression though since the current code is also not using AssertingIndexSearcher::search. I think we need a better integration of the `ContextIndexSearcher` if you're willing to keep the ability to extend `IndexSearcher` in a plugin but this is probably out of the scope of this pr that I consider as a workaround (hack ;)).